File: scala_study/src/main/scala/leetcode/deleteDuplicates.scala
package leetcode

class ListNode(_x: Int, _next: ListNode) {
  var x = _x
  var next = _next

}

object Solution{
  def deleteDuplicates(head: ListNode):ListNode = {
    if (head == null) return head

    var ans: ListNode = new ListNode(0,head)
    var start = ans

    while (start.next != null && start.next.next != null){
      if (start.next.x == start.next.next.x) {
        val X  = start.next.x
        while(start.next != null && start.next.x == X ) start.next = start.next.next
      }else{
        start = start.next
      }
    }
    ans.next
  }
}

File: JavaLearn/src/com/leetcode/DeleteDuplicates.java
package com.leetcode;

public class DeleteDuplicates {
    public ListNode deleteDuplicates(ListNode head){
        ListNode ans = new ListNode();

        ans.next =head;

        ListNode p = ans;

        while (p != null && p.next !=null && p.next.next != null){
            if(p.next.val == p.next.next.val){
                int x = p.next.val;
                while (p.next != null && p.next.val == x){
                    p.next = p.next.next;
                }
            }else {
                p = p.next;
            }
        }


        return ans.next;
    }
}
